src/session/mod.rs

```diff
@@ -252,6 +253,21 @@ impl SessionBuilder {
         self.connect_bio(tcp)
     }
 
+    pub fn connect_timeout<A>(
```

Comment:
IMHO it will have a better user experience if we combine the `connect` API & `connect_timeout` API since we already have a separate `timeout` API and it's better to make 'em consistent.
All customized requirements can on I/Os can already be implemented via the `connect_bio` API if the user wants to have two different timers.
What do you think of it?

Reply:
Added timeout to connect as an Option.